plugins/de.cau.cs.kieler.klighd/src/de/cau/cs/kieler/klighd/LightDiagramServices.java

```diff
@@ -245,6 +246,12 @@ public static void layoutDiagram(final LightDiagramLayoutConfig config) {
             final DiagramLayoutEngine engine = new KlighdLayoutSetup().getDiagramLayoutEngine();
             final IStatus status;
 
+            // Add interactive Layout run.
+            if (viewModel.getProperty(CoreOptions.INTERACTIVE_LAYOUT) ||
+                    (!viewModel.getChildren().isEmpty() && viewModel.getChildren().get(0).getProperty(CoreOptions.INTERACTIVE_LAYOUT))) {
+                layoutParameters.addLayoutRun(new InteractiveLayoutConfigurator());
+            }
+            
```

Comment:
I just stumbled over this change draft, and I'm kind of concerned about that.
I would prefer to not hard-code this setting in this fundamental class, but use the `getAdditionalLayoutConfigs()` hook of `AbstractDiagramSynthesis`, see lines 232-239 above.

Reply:
So you suggest to hard code this in the `AbstractDiagramSynthesis` instead?

Reply:
But I guess this can produce problems if a synthesis overrides the `getAdditionalConfigs()` doesn't it?

Reply:
Nope, not in `AbstractDiagramSynthesis` but in your specific diagram synthesis implementation, or at least in some subclass of `AbstractDiagramSynthesis` that is inherited by your diagram synthesis impl than.

Reply:
I am not sure whether this is possible. Normally a synthesis has no setup method which could check the viewModel and decide whether an interactive layout run should be added.
I could of course add the viewModel to `getAdditionalConfigs` as a parameter. Whether interactive layout should be invoked is not really synthesis specific but depends on the used layout algorithm.
Moreover, doing this for every synthesis would result in duplicate code.

Reply:
I suggest to add the viewModel as a parameter and change `getAdditionalLayoutConfigs` in the `AbstractDiargamSynthesis`. Doig so does not harm, since this does nothing if the `interactiveLayout` property is not set. This property is inherently different from the `interactive` property which activates interactive layout for some algorithms.